src/components/TextField/TextField.tsx

```diff
@@ -174,6 +230,8 @@ export interface TextFieldProps
   label?: BaseFormElement['label'];
   secondaryLabel?: string;
   tooltipMessage?: string;
+  startAdornment?: React.ReactNode;
```

Comment:
@ynotdraw and I tried typing these to only be specific components (`Icon`, `IconButton` and `Text`) but didn't have any luck.

It looks like[ MUI is typing theirs](https://github.com/mui-org/material-ui/blob/d6ea629e6babc7de5b6a1ccf0effcd360eb6f37f/packages/material-ui/src/InputAdornment/InputAdornment.d.ts#L16) as `React.ReactNode` as well. They just have a comment that states: 
```
    /**
     * The content of the component, normally an `IconButton` or string.
     */
```

I also checked the [Typescript cheatsheet ](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/basic_type_example) and it also says `React.ReactNode`.

In our Storybook docs you can see that I added it is recommended that they use Icon, IconButton, or Text. Does anyone feel like I need to add a comment here as well so if anyone looks at the source code they will see those recommendations? 🤷‍♀️